core/auth/src/main/java/software/amazon/awssdk/auth/credentials/CredentialUtils.java

```diff
@@ -28,6 +31,72 @@ private CredentialUtils() {
      * authenticate themselves.
      */
     public static boolean isAnonymous(AwsCredentials credentials) {
+        return isAnonymous((AwsCredentialsIdentity) credentials);
+    }
+
+    /**
+     * Determine whether the provided credentials are anonymous credentials, indicating that the customer is not attempting to
+     * authenticate themselves.
+     */
+    public static boolean isAnonymous(AwsCredentialsIdentity credentials) {
         return credentials.secretAccessKey() == null && credentials.accessKeyId() == null;
     }
+
+    /**
+     * Converts an {@link AwsCredentialsIdentity} to {@link AwsCredentials}.
+     *
+     * <p>Usage of the new AwsCredentialsIdentity type is preferred over AwsCredentials. But some places may need to still
+     * convert to the older AwsCredentials type to work with existing code.</p>
+     *
+     * <p>The conversion is only aware of {@link AwsCredentialsIdentity} and {@link AwsSessionCredentialsIdentity} types. If the
+     * input is another sub-type that has other properties, they are not carried over. i.e.,
+     * <ul>
+     *     <li>AwsSessionCredentialsIdentity -> AwsSessionCredentials</li>
+     *     <li>AwsCredentialsIdentity -> AwsBasicCredentials</li>
+     * </ul>
+     * </p>
+     *
+     * @param awsCredentialsIdentity The {@link AwsCredentialsIdentity} to convert
+     * @return The corresponding {@link AwsCredentials}
+     */
+    public static AwsCredentials toCredentials(AwsCredentialsIdentity awsCredentialsIdentity) {
+        if (awsCredentialsIdentity == null) {
+            return null;
+        }
+        // identity-spi defines 2 known types - AwsCredentialsIdentity and a sub-type AwsSessionCredentialsIdentity
+        if (awsCredentialsIdentity instanceof AwsSessionCredentialsIdentity) {
+            AwsSessionCredentialsIdentity awsSessionCredentialsIdentity = (AwsSessionCredentialsIdentity) awsCredentialsIdentity;
+            return AwsSessionCredentials.create(awsSessionCredentialsIdentity.accessKeyId(),
+                                                awsSessionCredentialsIdentity.secretAccessKey(),
+                                                awsSessionCredentialsIdentity.sessionToken());
+        }
+        if (isAnonymous(awsCredentialsIdentity)) {
+            return AwsBasicCredentials.ANONYMOUS_CREDENTIALS;
+        }
+        return AwsBasicCredentials.create(awsCredentialsIdentity.accessKeyId(),
+                                          awsCredentialsIdentity.secretAccessKey());
```

Comment:
Should we have `AwsCredentials` extend `AwsCredentialsIdentity` and `AwsSessionCredentials` extend `AwsSessionCredentialsIdentity` (if we don't already)?

If so, then should we do `if (awsCredentialsIdentity instanceof AwsCredentials) { return (AwsCredentials) awsCredentialsIdentity; }` to avoid copying the data for users of the old type?

Reply:
Good call! I was worried about extra conversions, but didn't think of checking here. Thanks for the suggestion. Will update.

Reply:
And they do extend already.

Reply:
Oh, adding similar check to `toCredentialsProvider` too.